x/auth/tx/builder.go

```diff
@@ -315,11 +315,14 @@ func (w *wrapper) GetTx() authsigning.Tx {
 	return w
 }
 
-// GetProtoTx returns the tx as a proto.Message.
-func (w *wrapper) AsAny() *codectypes.Any {
-	// We're sure here that w.tx is a proto.Message, so this will call
-	// codectypes.NewAnyWithValue under the hood.
-	return codectypes.UnsafePackAny(w.tx)
+func (w *wrapper) GetProtoTx() *tx.Tx {
+	return w.tx
+}
```

Comment:
I don't think a TxBuilder -> protoTx converter is ever needed.

As I see, we use `ProtoTxProvider` in 2 places:
- in BuildSimTx, for the tx simulate endpoint,
- in a test for said simulate endpoint.

That makes me think that maybe the simulate endpoint is not ideal:
https://github.com/cosmos/cosmos-sdk/blob/f602e9e9776ea37d5b1ac46b724cbf81dff4dd90/proto/cosmos/tx/v1beta1/service.proto#L14-L20

Maybe it should just be `tx_bytes` here, like in broadcast. Thoughts? cc @aaronc

Reply:
I prefer using a object types in proto messages rather than marshaled bytes. The whole point of message is to define structure with object types and let the protocol buffers do the binary / bytes rather than us hacking in bytes of object into the protocol.  This is more bullet / bug proof and forward looking for maintenance.

Reply:
Sounds good too 👍. In this case, could we make `ProtoTxProvider` somewhat local to tx simulations? Since that's the only place that TxBuilder -> protoTx conversion is needed.

Reply:
OK, moved to x/auth/tx